Build: Upgrade to Zig 0.15.1

npc1054657282 opened the pull request:
Currently all tests pass (Removed two tests that could not be restored, changed the expected result of a test, and changed the usage of several tests to workarounds). The log output to stderr is generally the same as in the previous version, but there are some differences due to changes in the default structure printing behavior.

Due to changes in how formatted strings are handled in zig 0.15.1, testing for adding formatting options for non-numeric printing is permanently broken, and there's no low-cost workaround.

Furthermore, changes to zig's default formatting for structure printing have forced some tests to modify their expected results. This change primarily affects logging results. To restore the original behavior, a future workaround might involve copying the zig standard library code from an older version.

The `stderr` writer now allows for a buffer. For simplicity, the unbuffered writer is still used.

npc1054657282 committed:
Build: Upgrade to Zig 0.15.1
Currently all tests pass (Removed a test that could not be restored,
changed the expected result of a test, and changed the usage of several
tests to workarounds). The log output to stderr is generally the
same as in the previous version, but there are some differences due to
changes in the default structure printing behavior.

Due to changes in how formatted strings are handled in
zig 0.15.1, testing for adding formatting options for non-numeric
printing is permanently broken, and there's no low-cost workaround.

Furthermore, changes to zig's default formatting for structure printing
have forced some tests to modify their expected results. This change
primarily affects logging results. To restore the original behavior,
a future workaround might involve copying the zig standard library code
from an older version.

npc1054657282 committed:
Fix: Unsafe undefined init in the test
Although it may seem that the undefined here is insignificant,
when logging, the undefined slice printed here will cause the
memory protection to be triggered.

npc1054657282 committed:
Refactor: Eliminate inconsistent behaviors
1. The parameter style of `StyleRecord.format` is inconsistent.
2. The behavior of some parts of `isOptional` and `isMultiple`
  in `fmt.formatNumber` is inconsistent with the original
  (specifically, `options` is incorrectly not inherited in `isOptional`,
  and in `isMultiple`, the logic for `begin`, `separator`, and `end`
  incorrectly inherits `options`).
  Eliminate the redundant duplications in `fmt.formatNumber`.

npc1054657282 committed:
Refactor: Distinguish intentional `.{}` options
1.In the internal implementation of `format`, for APIs that require
  `std.fmt.Options` parameters, some uses of `.{}` are intended
  by design, while others are due to the upgrade in Zig version 0.15.1,
  which prohibited format options of non-numeric types.
  There is a conscious distinction made between them.
2.Refactor `Range.format` and `Range.formatNumber` to make them use
  `Any.formatInner` directly.

npc1054657282 committed:
Doc: Modify the readme code
Modify the readme code that cannot be executed in zig 15.0.1 version.

kioz-wang commented on src/io/fmt.zig:
```diff
@@ -40,41 +40,110 @@ pub fn Any(V: type) type {
             return .{ .value = value, .options = options };
         }
-        pub fn format(self: Self, comptime fmt: []const u8, options: std.fmt.FormatOptions, writer: anytype) @TypeOf(writer).Error!void {
+        pub fn formatInner(self: Self, writer: *std.io.Writer, comptime fmode: u8, number: ?std.fmt.Number) std.io.Writer.Error!void {
```
我看了15.1对format的改动，几乎是重写了，同时增加了一些新的东西，比如对 optional、enum 等有了更合理的格式化实现。所以我需要更多的时间考虑这个 pr：这确实解决了编译的问题，但适配新版本并不只有编译的工作。是否要利用15.1中提供的特性来简化 zargs 的部分代码，这也是我想要考虑的

npc1054657282 replied:
我同意，但是我认为升级的第一步应该还是在还原原始行为逻辑，不引入新的改动的情况下的修改，更深化的修改是在确定正确性以后。虽然看起来代码修改量大，但大多只是api的兼容性变更，是一个最小化修改了。

kioz-wang replied:
你的想法是对的，升级包括编译适配（尽量保持原始行为逻辑）和适当的删减重构两个阶段，但是我不希望主线中引入冗余的代码。可以在单独的分支中完成两个阶段的工作，最后合并回主线。

kioz-wang commented:
` To restore the original behavior, a future workaround might involve copying the zig standard library code from an older version.`

不用考虑 workaround，不接受使用旧的 stdlib 代码，我希望这个项目始终跟上 zig 新的特性，直到 zig 1.0。如果 zig 发生重大变化，我会考虑重构 zargs 的部分代码。总之不要考虑任何会背上历史包袱的方案。

npc1054657282 committed:
Chore: Do not consider copy old stdlib code.
"不用考虑 workaround，不接受使用旧的 stdlib 代码，我希望这个项目始终跟上 zig
新的特性，直到 zig 1.0。如果 zig 发生重大变化，我会考虑重构 zargs 的部分代码。
总之不要考虑任何会背上历史包袱的方案。"

kioz-wang commented:
先合并到这个分支吧 81-update-to-zig-0151

kioz-wang commented:
> `In order to use the zig 0.15.1 version of zterm, I temporarily moved the zterm dependency to my personal fork.`
> 
> 请先修改 zterm 的 pr 让它合入到 feature/update-to-zig-0151，我会打一个 alpha tag，用来修改这个 dependency

use this

```shell
zig fetch --save https://github.com/kioz-wang/zterm/archive/refs/tags/v0.15.0-alpha.0.tar.gz
```

npc1054657282 committed:
Build: Modify dependency back to kioz-wang/zterm
Modify dependency to kioz-wang/zterm/v0.15.0-alpha.0

npc1054657282 commented:
已经修改dependency并将PR提交到新分支，关闭此PR
